replay/ann/ann_mixin.py

```diff
@@ -0,0 +1,195 @@
+import uuid
+from abc import abstractmethod
+from functools import cached_property
+from typing import Optional, Dict, Union, Any
+
+from pyspark.sql import DataFrame
+from pyspark.sql import functions as sf
+
```

Comment:
we need tests showing that the model works and provides the necessary amount of recommendations with ann inference. those tests also will also become an example of ann inference usage.